Entities/LobbyMapWarp.cs

```diff
@@ -0,0 +1,104 @@
+using Celeste.Mod.CollabUtils2.UI;
+using Celeste.Mod.Entities;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Monocle;
+using System.Collections;
+using System.Linq;
+
+namespace Celeste.Mod.CollabUtils2.Entities {
+    [CustomEntity("CollabUtils2/LobbyMapWarp")]
+    public class LobbyMapWarp : Entity {
+        private readonly string spritePath;
+        private readonly bool spriteFlipX;
+        private readonly string activateSpritePath;
+        private readonly bool activateSpriteFlipX;
+        private readonly Facings playerFacing;
+        
+        private LobbyMapController.MarkerInfo info;
+
+        private Sprite sprite;
+
+        public LobbyMapWarp(EntityData data, Vector2 offset) : base(data.Position + offset) {
+            spritePath = data.Attr("spritePath", "decals/1-forsakencity/bench_concrete");
+            spriteFlipX = data.Bool("spriteFlipX");
+            activateSpritePath = data.Attr("activateSpritePath", "CollabUtils2/characters/sitBench");
+            activateSpriteFlipX = data.Bool("activateSpriteFlipX");
+            playerFacing = (Facings) data.Int("playerFacing", (int) Facings.Right);
+            Depth = data.Int("depth", Depths.Below);
+
+            LobbyMapController.MarkerInfo.TryParse(data, null, out info);
+
+            if (!string.IsNullOrWhiteSpace(spritePath)) {
+                Add(sprite = new Sprite(GFX.Game, spritePath));
+                sprite.Effects = spriteFlipX ? SpriteEffects.FlipHorizontally : SpriteEffects.None;
+                sprite.Justify = new Vector2(0.5f, 1f);
+                sprite.Add("idle", string.Empty);
+                sprite.Play("idle");
+            }
+            
+            Add(new TalkComponent(new Rectangle(-16, -32, 32, 32), new Vector2(0, data.Float("interactOffsetY", -16f)), onTalk) {
+                PlayerMustBeFacing = false,
+            });
+        }
+
+        public override void Added(Scene scene) {
+            base.Added(scene);
+            var level = scene as Level;
+            info.SID = level.Session.Area.SID;
+            info.Room = level.Session.Level;
+        }
+
+        private void onTalk(Player player) {
+            if (player.Scene is Level level) {
+                if (!string.IsNullOrWhiteSpace(activateSpritePath)) {
+                    Add(new Coroutine(activateRoutine(player)));
+                } else {
+                    level.Add(new LobbyMapUI());
+                }
+            }
+        }
+
+        private IEnumerator activateRoutine(Player player) {
```

Comment:
this player animation setup still seems pretty unwieldy. might be more straightforward to just put the whole `sitBench` animation in the CU2 Sprites.xml and let people reskin it from XML if they want. having to read the docs and match naming schemes with hair textures feels messy (and doing it through XML would simplify the code quite a bit)

ideally we could just stick a new player animation in w/o reskinning all of `player` but that's a thing rn :(

this way you wouldn't have to worry about separate hair and player anims and would let you play the sprite directly from a bank

Reply:
Sounds good.